src/tuneflow_py/models/song.py

```diff
@@ -83,6 +85,89 @@ def remove_track(self, track_id: str):
                 dep_track.remove_output()
         return track
 
+    def get_structures(self):
+        return [StructureMarker(song=self, proto=structure_proto) for structure_proto in self._proto.structures]
+
+    def get_structure_at_index(self, index: int):
+        if index < 0:
+            index = 0
+
+        if index >= len(self._proto.structures):
+            index = len(self._proto.structures) - 1
```

Comment:
The above two checks are probably error-prone, the user might provide an out of range index (e.g. 999) and still get some structure which can lead to potential errors.

Note that this is different from get structure at tick since the index is more specific here.